### PR TITLE
[ME]: Add search filters 

### PR DESCRIPTION
**Pull request by tkohr**
### Description

This PR introduces a new `search-filters.component` for the md-editor app. It is displayed with a first filter allowing to filter on editor/user on the `allRecords` list. The component is also displayed on the `myRecords` list, but empty for now as the editor filter does not make much sense here. 

### Screenshots

image

### Quality Assurance Checklist

- [ ] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [ ] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

### CODE REVIEW

File: apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts [hunk 1]
```diff
@@ -313,6 +313,43 @@ describe('dashboard (authenticated)', () => {
...
+        cy.get('gn-ui-interactive-table')
+          .find('[data-cy="table-row"]')
+          .should('have.length', '5')
+        cy.get('gn-ui-results-table')
+          .find('[data-cy="table-row"]')
+          .first()
+          .as('record')
+        cy.get('@record')
+          .children('div')
+          .eq(4)
```
Comment (LHBruneton-C2C) [inline apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts:336]: This getting a Nth child of an element always feels a bit brittle to me. Can we easily add a data-cy attribute on the needed element here? There are other examples of data-cy in the code.

File: apps/metadata-editor/src/app/dashboard/search-filters/search-filters.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { SearchFiltersComponent } from './search-filters.component'
+import { MockBuilder } from 'ng-mocks'
+
+describe('SearchFiltersComponent', () => {
```
Comment (LHBruneton-C2C) [inline apps/metadata-editor/src/app/dashboard/search-filters/search-filters.component.spec.ts:5]: It's always a bit tricky to decide what to test, but I would at least test that a mock searchConfig with multiple elements is correctly read.

File: libs/feature/search/src/lib/utils/service/fields.spec.ts [hunk 3]
```diff
@@ -666,4 +686,41 @@ describe('search fields implementations', () => {
...
+        expect(repository.aggregate).toHaveBeenCalledWith({
+          'userinfo.keyword': {
+            type: 'terms',
+            limit: 1000,
+            field: 'userinfo.keyword',
+            sort: ['asc', 'key'],
+          },
+        })
+      })
+      it('returns the available editors, order by ascending key (alphabetical)', () => {
```
Comment (LHBruneton-C2C) [inline libs/feature/search/src/lib/utils/service/fields.spec.ts:708]: Looking at your new code, your new test is mostly testing the formatting of the user info. I'm a bit lost as to why we're talking of editors and sorting.

File: apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts (hunk 1 above)
Comment (tkohr) [reply apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts]: Yep, I've adapted accordingly. It didn't save that many lines, but has the advantage to easier test all the entries and not only the first :-)

File: apps/metadata-editor/src/app/dashboard/search-filters/search-filters.component.spec.ts (hunk 2 above)
Comment (tkohr) [reply apps/metadata-editor/src/app/dashboard/search-filters/search-filters.component.spec.ts:6]: You are right, I thought it was already covered by the e2e tests, but it is actually not. I've added a UT for it.

File: libs/feature/search/src/lib/utils/service/fields.spec.ts (hunk 3 above)
Comment (tkohr) [reply libs/feature/search/src/lib/utils/service/fields.spec.ts]: True, I've renamed the titles to make it clearer, the sorting is rather a side effect, indeed.

**Review by LHBruneton-C2C — APPROVED** [review]

**Pull request merged**